aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -79,4 +79,43 @@ Tensor _s_where_cpu(const Tensor& condition, const Tensor& self, const Tensor& o
   });
   return ret;
 }
+
+Tensor max_values(const Tensor& self, int64_t dim, bool keepdim) {
+  return std::get<0>(self.max(dim, keepdim));
+}
+
+Tensor min_values(const Tensor& self, int64_t dim, bool keepdim) {
+  return std::get<0>(self.min(dim, keepdim));
+}
+
+// argmax and argmin
+
+Tensor argmax(const Tensor& self, int64_t dim, bool keepdim) {
+  return std::get<1>(self.max(dim, keepdim));
+}
+
+Tensor argmax(const Tensor& self) {
+  return std::get<1>(self.contiguous().view({-1}).max(/*dim=*/0));
```

Comment:
nit: you could also use `self.reshape({-1})` I think

Reply:
Oh ok. Does it ensure the tensor is contiguous?